### PR TITLE
Added codesign and resources

### PR DESCRIPTION
**Pull request by stamepicmorg**
- [X] Added signing of windows binaries, resolves  https://github.com/rehlds/.github/issues/30

- [X] Added resource metainfo to windows binaries, resolves https://github.com/rehlds/.github/issues/29

- [X] Added signing for linux binaries, resolves https://github.com/rehlds/.github/issues/28

-------------

> [!CAUTION]  
> This is `infrastructure` update! **NO** code and  **NO** bugfixes was implemented at `ReHLDS` at this PR.

-------------

Also, I know that we have .rc files that are used to create binaries.

The problem is that we have not yet fully decided how their future will develop from a development point of view. Also, now there are only 2 .rc files that are built into hlds.exe and hltv.exe. All other dlls are built naked, which excites antiviruses.

For now, I have chosen to use [rcedit](https://github.com/electron/rcedit), written by the Electron project, which can successfully read and edit resources in compiled files. Which is what I am actually doing.

For now, I suggest leaving it as is. To launch the mechanism. And optimize it in subsequent iterations of development, to move the progress forward a little.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -78,7 +116,7 @@ jobs:
 
   testdemos:
     name: 'Test demos'
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
```
[the remark this answers is not in the document]
Comment (stamepicmorg) [reply .github/workflows/build.yml:119]: image

just -latest tag leads to an outdated version. if you specify 24, then it is a little newer and it is also LTS

it is not critical. i could roll-back it.

**Comment by stamepicmorg** [comment]
Tonight i will update PR. 

> [!TIP]  
> We collectively came to the conclusion that signing each file inside the archive with .asc files is a bit redundant. For now, we'll just leave the signatures of the final archive and the native Windows ones.

File: .github/workflows/build.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (a1batross) [reply .github/workflows/build.yml:119]: I heard that GitHub is planning to move ubuntu-latest to ubuntu-24.04 early next year or so.

[the remark this answers is not in the document]
Comment (stamepicmorg) [reply .github/workflows/build.yml:119]: yep, and i think it very slow to wait. why not to update to next LTS now? it stable.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -47,6 +47,11 @@ jobs:
           msbuild ${{ env.solution }} -p:Configuration="${{ env.buildRelease }}" /t:Clean,Build /p:Platform=${{ env.buildPlatform }} /p:PlatformToolset=v140_xp /p:XPDeprecationWarning=false
           msbuild ${{ env.solution }} -p:Configuration="${{ env.buildReleasePlay }}" /t:Clean,Build /p:Platform=${{ env.buildPlatform }} /p:PlatformToolset=v140_xp /p:XPDeprecationWarning=false
 
+      - name: Get rcedit from chocolatey
+        run: |
+          choco install rcedit -y
+        shell: "pwsh"
```
[the remark this answers is not in the document]
Comment (stamepicmorg) [reply .github/workflows/build.yml:53]: a im not trust to git bash console at windows. PowerShell is more native.

File: .github/workflows/build.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (SergeyShorokhov) [reply .github/workflows/build.yml:119]: from:
- https://github.com/actions/runner-images/issues/10636

> Rollout will begin on December 5th and will complete on **January 17th, 2025**

**Review by SergeyShorokhov — APPROVED** [review]

**Review by SergeyShorokhov — APPROVED** [review]

**Pull request merged**